Masonry/View.jsx

```diff
@@ -103,13 +104,21 @@ export default class Masonry extends React.PureComponent {
       </div>
     ),
     scrollAnchor: window,
-    threshold: window.innerHeight * 2
+    threshold: window.innerHeight * 2,
+    columnGutter: 0
   }
 
   state = { averageHeight: 300, pages: [] }
 
+  static getDerivedStateFromProps (nextProps, prevProps) {
+    if (!prevProps.props || nextProps.items.length !== prevProps.props.items.length) {
+      return {props: nextProps}
+    }
+    return {props: null}
+  }
+
   componentDidMount() {
-    this.layout(this.props);
+    this.layout(this.state.props || this.props);
```

Comment:
Where is `state.props`? I'm not seeing it set anywhere :)

Reply:
It's the `props returned in the `getDerivedStateFromProps`, what ever is returned from that static function is updated onto the state.

Reply:
Oh I see!

This needs to change before merging in order to optimize the memory consumption. I'm wondering if this is necessary given `componentDidUpdate` kind of does the same thing that `componentWillReceiveProps` did?

I'm guessing since React v16 changed the way this behaves?

Reply:
Ok. This was done to replace the deprecated `componentWillReceiveProps` that was in there before, but I can see if I can find time look into a different solution later (need to get into the office, I'm late.. :) ). I've pushed changes addressing the other comments for now.

Don't hesitate to come with pointers &&|| suggestions re: `componentWillReceiveProps => !getDerivedStateFromProps `. I didn't get the component to load properly without this, but did not investigate thoroughly after researching this solution.

Reply:
Actually, I just removed it now, and it works fine without it! I'll look into it..!

Reply:
So I removed the `getDerivedStateFromProps` func, as well as sending the `columnWidth` as an argument into the `getSelectorHeight` (needed). I'm using the module as is now in my project, but I haven't done a whole lot of testing (have been on vacation), but so far it looks good on my end.

Reply:
Thanks @jnaO the changes look great. 

By any chance do you have the project or demo to share? I'd love to see how it looks before we merge :)

Reply:
Thanks! 

Nowhere to show it off as of yet. Best guess is you'll have to wait for probably a month or so before my implementation of it goes public. I'll share a link after it does.